composer/lib/dependabot/composer/file_parser.rb

```diff
@@ -165,10 +165,19 @@ def git_dependency_details(package_details, requirement)
       end
 
       def lockfile_details(name:, type:)
-        key = lockfile_key(type)
-        parsed_lockfile.fetch(key, []).find { |d| d["name"] == name }
-      end
-
+        begin
+          key = lockfile_key(type)
+        rescue
+          key = nil
+        end
+   
```

Comment:
can we rescue everything in here instead of doing all the checks? Or maybe move up to the callers of `lockfile_details` and throw the `DependencyFileNotParseable` there?